### PR TITLE
text-style and compose-space fixes

### PR DESCRIPTION
**Pull request by dkochmanski**
This supersedes the previous PR https://github.com/McCLIM/McCLIM/pull/655.

### CODE REVIEW

File: Backends/common/fonts.lisp [hunk 1]
```diff
@@ -187,10 +187,13 @@ of letters specified in a separate kerning-table."))
...
-      (round (max size 2))
-      (or (getf +font-sizes+ size size)
-          (getf +font-sizes+ :normal))))
+  (cond ((numberp size)
+         (round (max size 2)))
+        ((null size)
+         (getf +font-sizes+ (text-style-size *default-text-style*)))
+        (T
+         (or (getf +font-sizes+ size size)
+             (error "~s is not a valid text style size!" size)))))
```
Comment (scymtym) [inline Backends/common/fonts.lisp]: How does this work? In this leg of the `cond`, `size` is not null, thus `(getf +font-sizes+ size size)` should not return null.

File: Core/clim-core/gadgets.lisp [hunk 2]
```diff
@@ -2635,8 +2635,8 @@ if INVOKE-CALLBACK is given."))
           (let* ((topmost-pane (if scroll-p
                                   ;list-pane
                                   (scrolling (:scroll-bar :vertical
-                                              :suggest-height height   ;; Doesn't appear to be working..
-                                              :suggest-width (if scroll-p (+ 30 (bounding-rectangle-width list-pane))))
+                                              :suggested-height height   ;; Doesn't appear to be working..
```
Comment (scymtym) [inline Core/clim-core/gadgets.lisp]: Is the comment still true after fixing the typo?

Comment (dkochmanski) [reply Core/clim-core/gadgets.lisp]: I don't know how to verify that (I didn't investigate what the actual code does – I've just picked the change from the original commit). Do you want me to remove the comment without testing?

File: Backends/common/fonts.lisp (hunk 1 above)
Comment (dkochmanski) [reply Backends/common/fonts.lisp]: you are right, fixed in the force-pushed version.

Comment (scymtym) [reply Backends/common/fonts.lisp]: `getf` defaults to `nil` so the third argument could be omitted. Not super important.

File: Core/clim-core/gadgets.lisp (hunk 2 above)
Comment (scymtym) [reply Core/clim-core/gadgets.lisp]: Maybe replace with "TODO investigate whether these arguments actually have an effect"

File: Backends/common/fonts.lisp (hunk 1 above)
Comment (dkochmanski) [reply Backends/common/fonts.lisp]: I want to put emphasis on returning NIL if not found

File: Core/clim-core/gadgets.lisp (hunk 2 above)
Comment (dkochmanski) [reply Core/clim-core/gadgets.lisp]: added comment as suggested

**Pull request merged**